L/libpolymake_julia/build_tarballs.jl

```diff
@@ -0,0 +1,57 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+import Pkg: PackageSpec
+
+const name = "libpolymake_julia"
+const version = v"0.1.0"
+
+# Collection of sources required to build libpolymake_julia
+const sources = [
+    ArchiveSource("https://github.com/oscar-system/libpolymake-julia/archive/v$(version).tar.gz",
+                  "533237a1a1f7ff544017b9576704b2f05c6e3139443ea0aaa20bc88560675e9a"),
+]
+
+# Bash recipe for building across all platforms
+const script = raw"""
+# remove $libdir from LD_LIBRARY_PATH as this causes issues with perl
+if [[ -n "$LD_LIBRARY_PATH" ]]; then
+LD_LIBRARY_PATH=$(echo -n $LD_LIBRARY_PATH | sed -e "s|[:^]$libdir\w*|:|g")
+fi
+
+cmake libpolymake-j*/ -B build \
+   -DJulia_PREFIX="$prefix" \
+   -DCMAKE_INSTALL_PREFIX="$prefix" \
+   -DCMAKE_FIND_ROOT_PATH="$prefix" \
+   -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}" \
+   -DCMAKE_BUILD_TYPE=Release
+
+VERBOSE=ON cmake --build build --config Release --target install -- -j${nproc}
+
+install_license $WORKSPACE/srcdir/libpolymake-j*/LICENSE.md
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+const platforms = expand_cxxstring_abis([
+    Linux(:x86_64; libc=:glibc, compiler_abi=CompilerABI(cxxstring_abi=:cxx11)),
+    MacOS(:x86_64; compiler_abi=CompilerABI(cxxstring_abi=:cxx11)),
+])
+
+# The products that we will ensure are always built
+const products = [
+    ExecutableProduct("polymake_run_script", :polymake_run_script),
+    LibraryProduct("libpolymake_julia", :libpolymake_julia),
+    FileProduct("share/libpolymake_julia/type_translator.jl",:type_translator),
+]
+
+# Dependencies that must be installed before this package can be built
+const dependencies = [
+    Dependency("CompilerSupportLibraries_jll"),
+    BuildDependency(PackageSpec(name="Julia_jll", version="v1.4.1")),
```

Comment:
Is there a particular reason not to use 1.4.2 here? Note that 1.4.1 has bad headers for the macOS version, see https://github.com/JuliaPackaging/Yggdrasil/issues/1372

Reply:
I copied the version from `libcxxwrap-julia` to make sure to use the same version, but apart from that I don't have any strong opinions on that.

Reply:
Well,1.4.2 is missing Windows binaries (why??? hopefully this will be better in the future?), which is why some packages don't use it. But since we don't really support Windows right now...

Reply:
This really shouldn't make any difference as `libpolymake_julia` doesn't use julia headers or libjulia directly (and this doesn't create any compat entries). We only use `Julia_jll` indirectly via `libcxxwrap_julia`, but since `libcxxwrap_julia` seems to work fine with that 1.4.1 version I think we can just keep it there until `libcxxwrap_julia` switches to 1.5.

In some sense it should also be possible to automatically inherit a `BuildDependency` from another package, then we wouldn't need to specify julia at all here.

Reply:
As I just learned on the Julia slack, work is already underway to make the Julia version part of the "platform triple/tuple", so then all these concerns hopefully will be resolved and we can just have separate binaries for each Julia release series (1.3, 1.4, 1.5, etc.). See https://github.com/JuliaLang/julia/commit/217e67d6213883435db92a397a81f3ffe6a00eec#diff-d985676b7739e0f47a51d1f028ec6fedR741-R761